src/hal/lib/hal_lib.c

```diff
@@ -266,8 +266,9 @@ void hal_print_error(const char *fmt, ...)
     va_start(args, fmt);
     const char *prefix = "HAL error: ";
     strncpy(_hal_errmsg, prefix, sizeof(_hal_errmsg));
```

Comment:
Sorry the last statement is not correct since `HALPRINTBUFFERLEN` might be something else than `sizeof(_hal_errmsg)`.